accepted/00083-containerized-proxy.md

```diff
@@ -0,0 +1,87 @@
+
+It is believed implementing Proxy containerization will surface a subset of common problems and solutions also useful to implement a containerized Server in future.
+
+# Detailed design
+[design]: #detailed-design
+
+## Delivery
+  - Proxy is delivered as one container
+    - Updates are delivered via new images on a public Registry
+  - Documentation is produced to describe how this container can be run on popular container runtimes: [docker](https://docs.docker.com/engine/reference/commandline/cli/) and [podman](https://podman.io/)
+  - A minimal [Helm](https://helm.sh/) chart is delivered and tested against [k3s](https://k3s.io/)
+
+## Onboarding
+  - Containerized Proxy is not a managed client, does not run `salt-minion`, is not registered in the usual way
+  - A new UI page (and API endpoint) allows to add containerized Proxies by specifying their FQDN
+    - After confirmation a System entry (in the System list) is created 
+      - Normal client actions (package management, configuration file management...) will not be available on that System
+      - Proxy-specific Server UI/API functionality, like showing the list of attached clients, will be available like on regular Proxies
+    - for each such System, the Server generates a tarball ("configuration directory") that can be downloaded
+    - Proxy container needs the configuration directory to be mounted as a volume. After start, it is fully connected and operational
+
+## Operating
+  - Container will expose Salt (4505/4506) and https (443) ports
+  - Container will mount volumes:
+    - one with the configuration directory
+    - one with cached content (`/var/cache/squid/`, `/var/cache/rhn`)
+    - one with log files (`/var/log/apache2`, `/var/log/squid`, `salt-broker` log)
+  - Proxy Containers can be started/stopped/exchanged - as long as the configuration directory stays the same the Proxy identity, history and functionality will be preserved. Optionally, the cached content directory can also be preserved for optimal performance
+  - Creating bootstrap scripts from containerized Proxies will not be possible. `mgr-bootstrap --hostname=$PROXY_NAME` from the Server will still be possible
+  - SSL certificates can be handled in three ways:
+    1. using the same CA as the Server, as generated at Server installation time. A new SSL certificate for the Proxy is generated at "configuration tarball" creation time. This will be the default option
+    2. using the same CA as the Server, as generated by the user ("bring your own SSL certs"). A new SSL certificate for the Proxy has to be placed in the configuration directory by the user
+    3. leave SSL termination to the user's infrastructure altogether
+
+## Limitations
+  - Only ZeroMQ Salt clients are supported
+  - No monitoring functionality
```

Comment:
This limitation really bugs me. Monitoring is a weakness in SUSE imho, so disabling this would even make it worse. If this is planed to be fixed in a later RFC I am more then happy to resolve this point without further discussion.